cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -68,7 +66,7 @@ const LOG_TARGET: &str = "cirrus::executor";
 /// The implementation of the Cirrus `Executor`.
 pub struct Executor<Block: BlockT, Client, TransactionPool, Backend, E> {
 	// TODO: no longer used in executor, revisit this with ParachainBlockImport together.
-	parachain_consensus: Box<dyn ParachainConsensus>,
+	primary_chain_client: Arc<dyn HeaderBackend<PBlock>>,
```

Comment:
As mentioned in the TODO above the definition of `ParachainConsensus`, I expected it to be `PrimaryChainInterface` eventually which will support all the primary chain functions required on the executor side, aside from the primary client related function, there will be some other functions provided by `PrimaryChainInterface`, e.g., the RPC functionality that can be used to submit the unsigned extrinsic to a primary node directly. Hiding everything relating to the primary node under a trait is a good design IMHO, we could inherit it.

Reply:
I just wanted to eliminate unused code for now, we can add one back later, but I expect it to be a trait alias rather than trait implemented by a struct that contains the same thing internally.